clang/lib/CodeGen/CGLoopInfo.cpp

```diff
@@ -788,6 +797,20 @@ void LoopInfoStack::push(BasicBlock *Header, clang::ASTContext &Ctx,
     }
   }
 
+  // Translate 'loop attributes' arguments to equivalent Attr enums.
+  // It's being handled separately from LoopHintAttrs not to support
+  // legacy GNU attributes and pragma styles.
+  //
+  // For attribute code_align:
+  // n - 'llvm.loop.align i32 n' metadata will be emitted.
+  for (const auto *A : Attrs) {
```

Comment:
Aren't you separately ensuring we don't have duplicates?  If so, you can just use `llvm::find_if`

Reply:
The codegen part has been updated using https://github.com/llvm/llvm-project/pull/70762/commits/22f66bd0199300a92ea2aefc66a2640e79b734a4